src/partition/ReceivedBoundingBox.cpp

```diff
@@ -131,24 +99,83 @@ void ReceivedBoundingBox::computeBoundingBox()
     // master sends feedbackmap to other master
     com::CommunicateBoundingBox(_m2n->getMasterCommunication()).sendFeedbackMap(_feedbackMap,0);             
   }
+  else if (utils::MasterSlave::_slaveMode)
+  {
+    utils::MasterSlave::_communication->broadcast(_remoteParComSize, 0);
+
+    // initializing the _globalBB
+    for (int remoteRank = 0; remoteRank < _remoteParComSize; remoteRank++ )
+    {
+      _globalBB[remoteRank]= _bb;
+    }
+    
+    // receive _globalBB from master
+    com::CommunicateBoundingBox(utils::MasterSlave::_communication).broadcastReceiveBoundingBoxMap(_globalBB);
 
 
-  if (utils::MasterSlave::_masterMode)
-  {
-    for (int i = 0; i < _remoteParComSize; i++)
+    for (auto &other_rank: _globalBB)
     {
-      int counter=0;
-      _m2n->getMasterCommunication()->receive(counter, 0);
-      _vertexCounters.push_back(counter);
+      if (compareBoundingBox(_bb,other_rank.second)) {
+        _feedback.push_back(other_rank.first);            
+      }
     }   
-    utils::MasterSlave::_communication->broadcast(_vertexCounters);
-  }
-  else if (utils::MasterSlave::_slaveMode)
-  {    
-    utils::MasterSlave::_communication->broadcast(_vertexCounters, 0);  
+
+    //send feedback to master
+    utils::MasterSlave::_communication->send(_feedback, 0);     
   }  
 }
 
+bool ReceivedBoundingBox::compareBoundingBox(mesh::Mesh::BoundingBox currentBB, mesh::Mesh::BoundingBox receivedBB)
+{
+  //int sizeofBB = currentBB.size();
+  bool intersect=1;
+
+  for (int i=0; i < _dimensions; i++) {
+
+    if ((currentBB[i].first < receivedBB[i].first && currentBB[i].second < receivedBB[i].first) || (receivedBB[i].first < currentBB[i].first && receivedBB[i].second < currentBB[i].first) ) {
```

Comment:
Please add a comment for this line explaining what you do

Reply:
added!